brave-cxf3/src/main/java/com/github/kristofa/brave/cxf3/BraveCxfExecutorService.java

```diff
@@ -0,0 +1,157 @@
+package com.github.kristofa.brave.cxf3;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ServerSpan;
+import com.github.kristofa.brave.ServerSpanThreadBinder;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.phase.PhaseInterceptorChain;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.*;
+
+/**
+ * Wraps around {@link ExecutorService} and passes {@link ServerSpan}
+ * from cxf {@link org.apache.cxf.message.Exchange} context (if exists)
+ * to current {@link ServerSpanThreadBinder} during execution.
+ */
+public class BraveCxfExecutorService implements ExecutorService, Closeable {
+
+  private final ExecutorService delegate;
```

Comment:
You're right but it may lead to other problems. When FooService is singleton, Executor will be injected as ThreadLocalProxy it means that Executor will be available only 
on current Thread. When someone would want to use it to execute any other async processing inside it will fail.

```java
  public class FooService {

    @Context
    private Executor executor;

    @POST
    @Path("/foo")
    public void foo(@Suspended AsyncResponse response) {
      // executor is taken from ThreadLocal
      executor.execute(() -> {
      	// executor is not available.
        response.resume("foo " + client.bar());
      });
    }
  }
```
But taking into account your other comment. 
We could change BraveCxfExecutorService to AsyncContext (or something like this) and stop implementing any interface, we could also clean ServerSpan there.

```java
  public class FooService {

    @Context
    private AsyncContext asyncContext;

    @POST
    @Path("/foo")
    public void foo(@Suspended AsyncResponse response) {   
      asyncContext.wrap(() -> {
        executor.execute(() -> {
          response.resume("bar");
        });
      });
    }
  }
```